### PR TITLE
refactor(NODE-6610): replace feature flag symbols with internal properties

### PR DESCRIPTION
**Pull request by W-A-James**
#### What is changing?

- replace symbol options with string ones

##### Is there new documentation needed for these changes?

No

#### What is the motivation for this change?

Our perf testing supports configuration via a JSON string, switching to internal string options means we can use this system to enable the yet to be released features for perf testing.

### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: src/cmap/connection.ts [hunk 1]
```diff
@@ -889,7 +889,7 @@ export class CryptoConnection extends Connection {
 
     const decryptedResponse = responseType?.make(result) ?? deserialize(result, options);
 
-    if (autoEncrypter[kDecorateResult]) {
+    if (autoEncrypter.__decorateDecryptionResult) {
```
[the remark this answers is not in the document]
Comment (W-A-James) [reply src/cmap/connection.ts:892]: No, I'm going back through and cleaning this up. not ready for review yet 😅

File: test/unit/connection_string.test.ts [hunk 2]
```diff
@@ -748,43 +747,6 @@ describe('Connection String', function () {
...
-    });
-
-    it('should only exist if specified on options', () => {
-      const flag = Array.from(FEATURE_FLAGS)[0]; // grab a random supported flag
-      const client = new MongoClient('mongodb://iLoveJavaScript', { [flag]: true });
-      expect(client.s.options).to.have.property(flag, true);
-      expect(client.options).to.have.property(flag, true);
-    });
-
-    it('should support nullish values', () => {
```
Comment (aditi-khare-mongoDB) [inline test/unit/connection_string.test.ts:781]: Can we add in testing to check that the `___` properties support null-ish values, specifically for `__enableMongoLogger`? I don't see any tests that check what happens when `__enableMongoLogger` flag is `null.`  l

File: test/integration/node-specific/feature_flags.test.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { expect } from 'chai';
 import { MongoClient, SeverityLevel } from '../../mongodb';
 
 describe('Feature Flags', () => {
-  describe('@@mdb.skipPingOnConnect', () => {
+  describe('__skipPingOnConnect', () => {
```
Comment (aditi-khare-mongoDB) [inline test/integration/node-specific/feature_flags.test.ts:6]: This describe block and test file name  is inaccurate now that we're not using feature flags. Can we make the naming more accurate?

**Review by aditi-khare-mongoDB — APPROVED** [review]
